torch/_refs/nn/functional/__init__.py

```diff
@@ -357,6 +382,17 @@ def l1_loss(
     return _apply_loss_reduction(loss, reduction)
 
 
+# CompositeImplicitAutograd - don't register decomp
+def log_softmax(
+    a: TensorLikeType,
+    dim: Optional[int] = None,
+    _stacklevel: int = 3,  # for compat when using TorchRefsMode(strict=True)
+    dtype: Optional[torch.dtype] = None,
+) -> TensorLikeType:
+    check(dim is not None, lambda: "implicit dim not supported, use dim=X")
```

Comment:
Let's add a comment for these checks that explains the behavior is deprecated in regular PyTorch, and this error is to inform users how to update their calls (otherwise it seems like we would just not put a default value for `dim` in the parameters?)